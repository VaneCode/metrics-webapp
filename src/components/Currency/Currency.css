.currencyDiv {
  display: flex;
  padding: 0.5rem;
  justify-content: space-between;
  border: solid 1px black;
  height: 11.5rem;
}

.currencyIcon {
  align-self: center;
  flex: 50%;
  height: 10rem;
}

.icon {
  width: 100%;
  height: 100%;
  object-fit: cover;
  filter:
    invert(34%) sepia(61%) saturate(571%) hue-rotate(181deg)
    brightness(93%) contrast(88%);
}

.currencyTags {
  flex: 50%;
  display: flex;
  flex-direction: column;
  align-items: center;
  justify-items: center;
}
