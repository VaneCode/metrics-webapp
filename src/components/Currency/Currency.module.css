.currencyDiv {
  display: flex;
  flex-direction: column;
  padding: 1rem;
}

.currencyIcon {
  align-self: center;
  width: 10rem;
  height: 6rem;
}

.icon {
  width: 100%;
  height: 100%;
  object-fit: cover;
  filter: invert(34%) sepia(61%) saturate(571%) hue-rotate(181deg)
    brightness(93%) contrast(88%);
}

.currencyTags {
  height: 50%;
  display: flex;
  flex-direction: column;
  align-items: flex-end;
}
