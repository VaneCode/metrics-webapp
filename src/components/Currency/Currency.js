// Import
import PropTypes from 'prop-types';
import { useState } from 'react';
import './Currency.css';
import dollar from './dollar.png';

const Currency = (props) => {
  const [imgStatus, setImgStatus] = useState(true);
  const { currency } = props;
  return (
    <div className="currencyDiv">
      <div className="currencyIcon">
        <img
          src={imgStatus ? currency.icon : dollar}
          onError={() => setImgStatus(false)}
          alt="Currency icon"
          className="icon"
        />
      </div>
      <div className="currencyTags">
        <h3>{currency.currencyName}</h3>
        <span>{currency.currencyCode}</span>
      </div>
    </div>
  );
};

Currency.propTypes = {
  currency: PropTypes.shape({
    icon: PropTypes.string,
    currencyName: PropTypes.string,
    currencyCode: PropTypes.string,
  }).isRequired,
};

export default Currency;
