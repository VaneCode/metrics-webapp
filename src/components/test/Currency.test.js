import { Provider } from 'react-redux';
import TestRenderer from 'react-test-renderer';
import store from '../../redux/configureStore';
import Currency from '../Currency/Currency';

describe('Currency test', () => {
  test('render', () => {
    const currency = {
      currencyCode: 'AGLD',
      currencyName: 'Adventure Gold',
      icon:
        'https://currencyfreaks.com/photos/flags/agld.png',
    };
    const tree = TestRenderer.create(
      <Provider store={store}>
        <Currency currency={currency} />
      </Provider>,
    );
    expect(tree).toMatchSnapshot();
  });
});
