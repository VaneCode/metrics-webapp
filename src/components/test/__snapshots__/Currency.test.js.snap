// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Currency test render 1`] = `
<div
  className="currencyDiv"
>
  <div
    className="currencyIcon"
  >
    <img
      alt="Currency icon"
      className="icon"
      onError={[Function]}
      src="https://currencyfreaks.com/photos/flags/agld.png"
    />
  </div>
  <div
    className="currencyTags"
  >
    <h3>
      Adventure Gold
    </h3>
    <span>
      AGLD
    </span>
  </div>
</div>
`;
